src/agent/Core/ApplicationPool/Group/ProcessListManagement.cpp

```diff
@@ -88,13 +94,19 @@ Group::findProcessWithLowestBusyness(const ProcessList &processes) const {
 	}
 
 	int lowestBusyness = -1;
+    unsigned int highest_gen = 0;
 	Process *leastBusyProcess = NULL;
 	ProcessList::const_iterator it;
 	ProcessList::const_iterator end = processes.end();
 	for (it = processes.begin(); it != end; it++) {
 		Process *process = (*it).get();
 		int busyness = process->busyness();
-		if (lowestBusyness == -1 || lowestBusyness > busyness) {
+		if (lowestBusyness == -1 ||
+			lowestBusyness > busyness ||
+			(busyness == lowestBusyness && process->generation > highest_gen)) {
```

Comment:
I think there should not be a check for `enabledProcessBusynessLevels[i] == lowestBusyness` here. Check earlier comment.